### PR TITLE
Add option for alternative ratio calculation for VarVsVar

### PR DESCRIPTION
**Pull request by AndriusVaitkus97**
## Summary

This pull request introduces the following changes

* Fixed a bug where `ylabel_ratio` non-default value is overwritten by the default
* Added a new argument `ratio_method` to allow for alternative methods of calculating relative differences between lines in the plot. Added one extra method called `root_square_diff`, similar to the plot [here](https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/PAPERS/JETM-2022-01/fig_09a.png).

## Notes
* Variable names are not super intuitive - happy to hear other suggestions. Also technically `ratio` is not a correct term, but it is in line with the other plotting classes like `Histogram`, so I don't mind it staying.

### CODE REVIEW

File: puma/plot_base.py [hunk 1]
```diff
@@ -216,7 +216,13 @@ def __post_init__(self):
             [None] * self.n_ratio_panels if self.ymax_ratio is None else self.ymax_ratio
         )
 
-        self.ylabel_ratio = ["Ratio"] * self.n_ratio_panels
+        if self.ylabel_ratio is None:
+            self.ylabel_ratio = ["Ratio"] * self.n_ratio_panels
+        if len(self.ylabel_ratio) != self.n_ratio_panels:
+            raise ValueError(
+                f"If you passed `ylabel_ratio` of length {len(self.ylabel_ratio)}, "
+                f"but n_ratio_panels of {self.n_ratio_panels}"
```
Comment (samvanstroud) [inline puma/plot_base.py:224]: The error message doesn't make sense

**Review by samvanstroud — CHANGES_REQUESTED** [review]
Thanks @andriusvaitkus97, this looks good to me. I had one minor comment, but also it would be great if you could add a test case here:
https://github.com/umami-hep/puma/blob/main/puma/tests/utils/test_histogram_utils.py#L226. Cheers!

File: puma/plot_base.py (hunk 1 above)
Comment (AndriusVaitkus97) [reply puma/plot_base.py:224]: Sorry for a delay in replying, was busy with other stuff.

Yeah I messed up the error message from too much re-writing.
My logic was that if one provides _N_ ratio panel labels, but doesn't have _N_ ratio panels, it should not work. But looking at the rest of the `puma` code, I think multiple ration panels are only used for ROC curves so I can remove this check altogether.  For now I will just fix the error message.

File: puma/tests/utils/test_histogram_utils.py [hunk 2]
```diff
@@ -241,6 +241,16 @@ def setUp(self):
             0.05,
             0.025,
         ])
+        self.step_rsd = np.array([4, 4, -5.1961524, 0, -4.8989795, -8, -11.8321596])
+        self.step_rsd_unc = np.array([
+            0.625,
+            0.625,
+            0.5773503,
+            0,  # for RSD, if num=denom error, error is 0? or num_unc
```
Comment (AndriusVaitkus97) [inline puma/tests/utils/test_histogram_utils.py:249]: Question here more about the logic of uncertainty calculations. If there is an error associated with the numerator, but `num=denom`, should the error be zero or equal to `num_unc`?

**Comment by AndriusVaitkus97** [comment]
> Thanks @AndriusVaitkus97, this looks good to me. I had one minor comment, but also it would be great if you could add a test case here: https://github.com/umami-hep/puma/blob/main/puma/tests/utils/test_histogram_utils.py#L226. Cheers!

Done! I have one question/comment about it above

File: puma/tests/utils/test_histogram_utils.py (hunk 2 above)
Comment (samvanstroud) [reply puma/tests/utils/test_histogram_utils.py:249]: Good q, we had an issue about this https://github.com/umami-hep/puma/issues/213, which may have been resolved prematurely. I think if we are calculating the error on the ratio, we don't need to put an error on the nominal line at y=1 on the ratio panel.

Comment (samvanstroud) [reply puma/tests/utils/test_histogram_utils.py:249]: I reopened https://github.com/umami-hep/puma/issues/213, we can use that to follow up on this, I'm happy to merge this now

Comment (AndriusVaitkus97) [reply puma/tests/utils/test_histogram_utils.py:249]: Give me until he end of today please, I will double check with some JetEtMiss people about the RSD, whether we need to show the numerator uncertainty for the reference line (y=0). I guess for consistency with the ratio we shouldn't here either, but I wanted to check

Comment (AndriusVaitkus97) [reply puma/tests/utils/test_histogram_utils.py]: UPD: yeah 0 is fine for RSD @samvanstroud, from my side I think this is good to merge

**Review by samvanstroud — APPROVED** [review]

**Pull request merged**